### DISCUSSION TITLE
Add `system` output type

### DISCUSSION THREAD

**Discussion by jamesarosen** [web-infra-dev/rspack#3110] [Ideas]
WebPack has [built-in compatibility](https://github.com/systemjs/systemjs#compatibility-with-webpack) with [SystemJS](https://github.com/systemjs/systemjs). As of May 2023, rspack's `output.library.type` [supports many values](https://www.rspack.dev/config/output.html#outputlibrarytype), but not `system`.

SystemJS, while not standards-track, is widely used, especially in micro-frontend systems like [single-spa](https://single-spa.js.org/).

One caveat is that SystemJS imports are context-sensitive, which is why things like [systemjs-webpack-interop](https://github.com/joeldenning/systemjs-webpack-interop) exist.

If the rspack team would prefer this as a plugin, it would be helpful to have a reference plugin for supporting a new `output.libraryTarget` (or `output.library.type`).

**Comment by jamesarosen** [comment]
For now, a workaround is to use `libraryTarget: "amd"` and SystemJS's AMD plugin.

**Comment by hardfist** [comment]
yes system format is widely used in micro-frontend, we may support it in the future, I create a issue to be tracked here https://github.com/web-infra-dev/rspack/issues/3112

**Reply by matthewlee-toast** [reply]
is this be resolved? I see the linked issues has been resolved + merged

**Reply by hardfist** [reply]
i think this is supported now
